src/citrine/builders/auto_configure.py

```diff
@@ -0,0 +1,562 @@
+            self,
+            *,
+            table: GemTable,
+            mode: AutoConfigureMode,
+            print_status_info: bool
+    ):
+        """Create a default predictor from a GEM table."""
+        print("Configuring default predictor from GEM table...")
+        data_source = GemTableDataSource(table_id=table.uid, table_version=table.version)
+        predictor = self.project.predictors.auto_configure(
+            training_data=data_source, pattern=mode.value  # Uses same string pattern
+        )
+        predictor.name = f'{self._prefix}Default Predictor'
+        predictor = self.project.predictors.register(predictor)
+        predictor = wait_while_validating(
+            collection=self.project.predictors, module=predictor,
+            print_status_info=print_status_info
+        )
+
+        self._predictor = predictor
+        self._status = AutoConfigureStatus.PREDICTOR_CREATED
+        self._status_info = predictor.status_info
+
+        if predictor.status == 'INVALID':
+            self._status = AutoConfigureStatus.PREDICTOR_INVALID
+            raise RuntimeError("Predictor is invalid,"
+                               " cannot proceed to design space configuration.")
+
+    def _predictor_evaluation_stage(
+            self,
+            *,
+            predictor: Predictor,
+            evaluator: Optional[PredictorEvaluator],
+            print_status_info: bool
+    ):
+        """Evaluate the predictor with the specified evaluator, or create a default workflow."""
+        print("Configuring predictor evaluation workflow/execution...")
+        if evaluator is None:
+            # No evaluator, so create a default PEW for the predictor
+            pew = self.project.predictor_evaluation_workflows.create_default(
+                predictor_id=predictor.uid
+            )
+        else:
+            # We got an evaluator, so make a new PEW and register it manually
+            pew = PredictorEvaluationWorkflow(
+                name=f'{self._prefix}Predictor Evaluation Workflow',
+                evaluators=[evaluator]
+            )
+            pew = self.project.predictor_evaluation_workflows.register(pew)
+        pew = wait_while_validating(
+            collection=self.project.predictor_evaluation_workflows, module=pew,
+            print_status_info=print_status_info
+        )
+
+        self._predictor_evaluation_workflow = pew
+        self._status = AutoConfigureStatus.PEW_CREATED
+        self._status_info = pew.status_info
+
+        if pew.status == 'FAILED':
+            # Can proceed without raising error, but can't get PEE
+            self._status = AutoConfigureStatus.PEW_FAILED
+            warnings.warn("Predictor evaluation workflow failed -- unable to configure execution.")
+        else:
+            if evaluator is None:
+                # Get execution from default
+                self._predictor_evaluation_execution = next(pew.executions.list(), None)
```

Comment:
The module validation completion and execution creation aren't guaranteed to occur in any particular order, so there's a chance for a small race condition here where the PEW is valid but the execution hasn't been created.  They occur pretty tightly together, so a short wait here might be appropriate.  Alternatively, you could keep listing until an execution is present, up to a timeout.

Reply:
added a 5 sec timeout on PEE List